docs/20_smart-contracts/40_guides/10_resources_guide.md

```diff
@@ -0,0 +1,232 @@
+
+* The smart contract can not store any additional information if it consumes all its allocated RAM. To continue to save data in the blockchain database, one, or both of the following conditions must be met:
+
+  * A portion of the occupied RAM is freed by the smart contract.
+  * More RAM is allocated to the smart contract account through the RAM purchase process.
+
+### How To Purchase RAM
+
+The RAM resource must be bought with the `EOS` system token. The price of RAM is calculated according to the unique Bancor liquidity algorithm which is implemented in the system contract [here](https://docs.eosnetwork.com/system-contracts/latest/reference/Classes/structeosiosystem_1_1exchange__state).
+
+The quickest way to calculate the price of RAM:
+
+1. Run the following dune command, make sure you run it against the mainnet or the testnet of your choice:
+
+    ```shell
+    dune -- cleos get table eosio eosio rammarket
+    ```
+
+2. Observe the output which should look like the sample below:  
+
+    ```text
+    {
+        "supply": "10000000000.0000 RAMCORE",
+        "base": {
+            "balance": "35044821247 RAM",
+            "weight": "0.50000000000000000"
+        },
+        "quote": {
+            "balance": "3158350.8754 EOS",
+            "weight": "0.50000000000000000"
+        }
+    }
+    ```
+
+3. Make note of the `base balance`, in this case is 35044821247.
+4. Make note of the `quote balance`, in this case is 3158350.8754.
+5. Calculate the price of 1Kib of RAM as `quote balance` * 1024 / `base balance` = 0.0922 EOS.
+
+#### Buy RAM With Command Line Interface
+
+You can buy RAM through dune command line interface tool. And you can buy either an explicit amount of RAM expressed in bytes or an amount of RAM worth of an explicit amount of EOS.
+
+#### Buy RAM In EOS
+
+For example the below command buys for account `bob` 0.1 EOS worth of RAM at the current market RAM price. The cost for the RAM and the execution of this transaction is covered by the `alice` account and the transaction is authorized by the `active` key of the `alice` account.
+
+```shell
+dune -- cleos system buyram alice bob "0.1 EOS" -p alice@active
+```
+
+#### Buy RAM In Bytes
+
+For example the below command buys for account `bob` 1000 RAM bytes at the current market RAM price. The cost for the RAM and the execution of this transaction is covered by the `alice` account and the transaction is authorized by the `active` key of the `alice` account.
+
+```shell
+dune -- cleos system buyrambytes alice bob "1000" -p alice@active
+```
+
+#### Buy RAM With EOS Wallet
```

Comment:
Since we're focusing on local development for all guides, let's not discuss wallets at all outside of things like specific wallet guides and front-end integration guides.